java/vector/src/test/java/org/apache/arrow/vector/types/pojo/TestExtensionType.java

```diff
@@ -342,6 +347,34 @@ public void set(int index, UUID uuid) {
       bb.putLong(uuid.getLeastSignificantBits());
       getUnderlyingVector().set(index, bb.array());
     }
+
+    @Override
+    public TransferPair makeTransferPair(ValueVector to) {
```

Comment:
This `TransferPair` is likely a very common pattern for extension types, so one could create a concrete `ExtensionWriter` which actually implements this and something like `UuidWriter` could inherit from. 

We do something similar for our writers: https://github.com/xtdb/xtdb/blob/e9a15ee12244367c1d3d1a75c00e4f5c0e10008f/core/src/main/kotlin/xtdb/vector/extensions/XtExtensionVector.kt#L30-L40

Reply:
Seems to be fine, let's get this one working and further evaluate.